### DISCUSSION TITLE
closing stale discussions ?

### DISCUSSION THREAD

**Discussion by orbeckst** [MDAnalysis/mdanalysis#4622] [General]
Does it make sense to close discussions that have gone stale?

In particular I am thinking of any discussion where a question is asked, someone replies with a request for more information and nothing comes back from the original poster.

I'd expect someone with a problem to keep an eye on a discussion and engage. If they solve their own problem then it would be fantastic if they posted their own solution. If for whatever reason they are not anymore interested in an answer that's also fine but then we should make clear that nobody else should be spending their precious time on dealing with the question.

I would be interested in hearing from anyone in the forum if there are reasons to keep everything open. 

If we want to be able to close stale discussions then we need to have guidelines what "stale" means and how long it takes until a discussion become stale.

## Definition of a stale discussion
My definition would be: *A discussion is **stale** when a question was asked of the original poster and <strike>no substantive answer</strike> the original poster has not replied to requests for clarifications or more information <strike>was posted</strike> within 2 weeks and there are no other ongoing active discussions (posts on a near-daily basis) happening in the thread.**

## Closing stale discussions
I would then say that anyone with the appropriate permissions should feel empowered to close any stale discussion when they come across one. If there's value in leaving it open, post a short comment to the effect and say why — e.g. the post demonstrates a more widespread issue or there's a discussion that is of broad value etc.

EDIT: Clarified my definition, based on feedback in comments.

**Comment by orbeckst** [comment]
Of course, don't close any Q&A that **answered** a question — these are very valuable as a resource!

**Comment by RMeli** [comment]
I think it would be good to have a clear policy on this. I've closed some discussions to clean up, mainly because by default only the "Open" ones were displayed (this seems to have changed recently, where by default "All" are displayed).

The discussions I've closed were mostly introductions by members of the community. While it's fantastic to hear from the members of our community, once pleasantries are exchanged there is not much benefit for other members to have those "Open" and mixed with information about MDAnalysis.

> In particular I am thinking of any discussion where a question is asked, someone replies with a request for more information and nothing comes back from the original poster.

I think this also warrants closure, to clarify that the discussion has gone stale and does not contain useful information. However, there might be cases where discussions go stale after a few (or many) exchanges, and the exchange does contain valuable information. I think we should distinguish the two and leave "Open" the latter.

The same for Q&A. Q&A with answered questions or interesting information should remain open. Q&A that go stale without useful information should be "Closed".

> A discussion is stale when a question was asked of the original poster and no substantive answer was posted within 2 weeks and there are no other ongoing active discussions (posts on a near-daily basis) happening in the thread.

I don't fully agree with this definition. "No substantive answer" also includes "no answer at all", and this is unfair on whoever asked the question/started the discussion. I think we should strive to have at least one answer to each question/discussion no matter how old.

I'm saying this because I recently answered some old questions that we did not immediately acknowledge (sorry about that!) and got some reactions from the original poster (either engagement, or a thumb up).

I'd define a stale discussion as something with un-answered follow-up questions from the person trying to answer, or discussion with a back and forth that died down and does not contain any useful information/resolution. (But then we will have to decide what's useful...)

> I would then say that anyone with the appropriate permissions should feel empowered to close any stale discussion when they come across one. If there's value in leaving it open, post a short comment to the effect and say why — e.g. the post demonstrates a more widespread issue or there's a discussion that is of broad value etc.

+1 for this approach.

**Reply by RMeli** [reply]
This is a discussion that I would consider stale if I don't get an answer in a couple of weeks: https://github.com/MDAnalysis/mdanalysis/discussions/4611. Although it took me two weeks to answer, so I don't certainly expect a fast reply back.

**Reply by orbeckst** [reply]
Sorry, I phrased the following very poorly by using "answer" when I really meant "additional information"
> <strike>A discussion is stale when a question was asked of the original poster and no substantive answer was posted within 2 weeks and there are no other ongoing active discussions (posts on a near-daily basis) happening in the thread.</strike>

I wanted to say

  A discussion is stale when a question was asked of the original poster and the original poster has not replied to requests for clarifications or more information within 2 weeks and there are no other ongoing active discussions (posts on a near-daily basis) happening in the thread.

I'd say that clearly defined problems without answers are useful starting points and I would leave them open.

**Reply by orbeckst** [reply]
I agree with you on discussion #4611 , that was one of the examples I had in mind.

**Reply by RMeli** [reply]
> A discussion is stale when a question was asked of the original poster and the original poster has not replied to requests for clarifications or more information within 2 weeks and there are no other ongoing active discussions (posts on a near-daily basis) happening in the thread.

This is what I had in mind too, so sounds great for me.
